### ISSUE TITLE
Vanilla sprite rendering anomalies

### ISSUE THREAD

**Issue by lemming104** [Helion-Engine/Helion#584]
I've been playing Helion (currently on 273cddc) with the vanilla sprite rendering mode turned on, and the results mostly look really good.  Since I'm running this on a system with a very powerful GPU, I'm not even noticing much of a performance hit.  I have noticed a couple of interesting anomalies here and there, though.  

Both of these ones are from Doom 2, MAP02:

![helion_20240719_11 03 29 2783](https://github.com/user-attachments/assets/a0b41d17-d864-4665-943a-7775168d73e6)
![helion_20240719_11 14 11 8433](https://github.com/user-attachments/assets/9c3fd822-8379-42d7-a079-a39f051caeea)

**Comment by nstlaurent** [comment]
This case is a catch 22 and happens because the sprites are never discarded to the floor/ceiling. The fix would be to discard them to floors and ceilings but then it's back to the way it was before. The barrel is really funny, had no idea there was this stray barrel not really in the map.

The real problem with the barrel appears to be that it's in the ceiling for some reason and it's supposed to render to the floor. An interesting point the software renderer only sometimes renders it like below:

![image](https://github.com/user-attachments/assets/5405517e-1791-4449-a558-dd0887d67eb2)

**Comment by lemming104** [comment]
ID Software really messed up that map in v1.9 in general, with stuff stuck in walls and whatnot.

Overall, the vanilla sprite rendering mode seems to work well; I played through all of Doom E1 and several levels of Eviternity with it on, and it's nice seeing a hardware renderer _not_ clip the tops off the sprites in the "tunnel maze" part of E1M4.

The only other quirk I've noticed so far is that the imps on the ledges at the end of E1M2 get their feet clipped a bit.
![helion_20240719_06 05 55 4257](https://github.com/user-attachments/assets/695f68b7-17db-4c40-9761-60cf6eb092c6)
